src/Protocol/ClientEncoder.php

```diff
@@ -24,18 +25,27 @@ public function encode($name, $args)
 
     public function __doRequest($request, $location, $action, $version, $one_way = 0)
     {
+        $finalLocation = $this->locationOverride !== null ? $this->locationOverride : $location;
+
         $this->request = new Request(
             'POST',
-            (string)$location,
+            (string) $finalLocation,
             new Headers(array(
-                'SOAPAction' => (string)$action,
+                'SOAPAction' => (string) $action,
                 'Content-Type' => 'text/xml; charset=utf-8',
                 'Content-Length' => strlen($request)
             )),
-            new Body((string)$request)
+            new Body((string) $request)
         );
 
         // do not actually block here, just pretend we're done...
         return '';
     }
+
+    public function overrideLocation($newLocation)
+    {
+        $copy = clone $this;
+        $this->locationOverride = $newLocation;
+        return $copy;
+    }
```

Comment:
PHP's SoapClient already offers a `__setLocation()` method, perhaps we can rely on this there instead of manually keeping track of the location? (see also above)